File: moxy/compiler/src/main/java/com/omegar/mvp/compiler/pipeline/Validator.java
package com.omegar.mvp.compiler.pipeline;


/**
 * Created by Anton Knyazev on 03.12.2020.
 */
abstract public class Validator<T> extends Processor<T, T> {

    @Override
    public void process(T input, PipelineContext<T> context) {
        if (validate(input)) {
            super.process(input, context);
        }
    }

    public abstract boolean validate(T input);

}

File: moxy/compiler/src/main/java/com/omegar/mvp/compiler/pipeline/Validator.kt
package com.omegar.mvp.compiler.pipeline

/**
 * Created by Anton Knyazev on 03.12.2020.
 */
abstract class Validator<T> : Processor<T, T>() {
    override fun process(input: T, context: PipelineContext<T>?) {
        if (validate(input)) {
            super.process(input, context)
        }
    }

    abstract fun validate(input: T): Boolean
}